modules/indexer/code/indexer.go

```diff
@@ -64,6 +64,16 @@ func Init() {
 	go func() {
 		start := time.Now()
 		log.Info("PID: %d Initializing Repository Indexer at: %s", os.Getpid(), setting.Indexer.RepoPath)
+		defer func() {
+			if err := recover(); err != nil {
+				log.Error("PANIC whilst initializing repository indexer: %v\nStacktrace: %s", err, log.Stack(2))
+				log.Error("The indexer files are likely corrupted and may need to be deleted")
+				cancel()
+				indexer.Close()
```

Comment:
Can this panic the panic?
```suggestion
				indexer.Close()
```

Reply:
As far as I can see no. Although it's very difficult to assess the code completely.